### DISCUSSION TITLE
Docker run --init flag using docker go client

### DISCUSSION THREAD

**Discussion by BhanuKiranChaluvadi** [moby/moby#43834] [Q&A]
Hi, 

I would like to accomplish  terminal command `docker run --init`  through go docker client and went through 
 [ContainerCreate](https://github.com/moby/moby/blob/master/client/container_create.go) and [ContainerStart](https://github.com/moby/moby/blob/master/client/container_start.go) there is an option for custom init  in [host_config](https://github.com/moby/moby/blob/master/api/types/container/host_config.go) . How to set the default init arguement like in `docker run --init` ?

Thank you

**Comment by cpuguy83** [comment]
The option in hostconfig is what needs to be set.
There is a default value which can be configured on the daemon.

**Reply by BhanuKiranChaluvadi** [reply]
So should I configure docker daemon to some think like this one - [daemon-configuration-file](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-configuration-file)  on host machine ?.  After the configuration all the docker container instances has --init argument as default.

**Reply by thaJeztah** [reply]
The daemon configuration indeed sets it as a default for all new containers. If you need it to be set for a specific container, it should be set in the hostconfig when creating the container. The Docker CLI code doing that can be found here; https://github.com/docker/cli/blob/4411b515f5e9798f371ca1300589fccd64fe5001/cli/command/container/opts.go#L675-L678
